Rather than using uname -m, call AC_CANONICAL_HOST and use host_cpu.

Upstream-status: pending
Signed-off-by: Christopher Larson <chris_larson@mentor.com>

--- oprofile-0.9.8.orig/configure.ac
+++ oprofile-0.9.8/configure.ac
@@ -14,6 +14,7 @@ AC_PREREQ(2.13)
 AC_INIT(libop/op_config.h)
 AM_INIT_AUTOMAKE(oprofile, 0.9.8)
 AM_CONFIG_HEADER(config.h)
+AC_CANONICAL_HOST
 
 AC_CHECK_DECLS([basename], [], [], [[#include <libgen.h>]])
 AC_PROG_RANLIB
@@ -157,8 +158,7 @@ AC_DEFINE_UNQUOTED(HAVE_PERF_EVENTS, $HA
 
 if test "$HAVE_PERF_EVENTS" = "1"; then
 	PFM_LIB=
-	arch="`uname -m`"
-	if test "$arch" = "ppc64" || test "$arch" = "ppc"; then
+	if test "$host_cpu" = "powerpc64" || test "$host_cpu" = "powerpc"; then
 		AC_CHECK_HEADER(perfmon/pfmlib.h,,[AC_MSG_ERROR([pfmlib.h not found; usually provided in papi devel package])])
 		AC_CHECK_LIB(pfm,pfm_get_event_name, HAVE_LIBPFM3='1'; HAVE_LIBPFM='1', [
 			AC_CHECK_LIB(pfm,pfm_get_os_event_encoding, HAVE_LIBPFM3='0'; HAVE_LIBPFM='1',
